### ISSUE TITLE
Piece.updateCellsEmitter

### ISSUE THREAD

**Issue by jonathanolson** [phetsims/fractions-intro#55]
```js
    // create emitter that will signal the the piece has reached its cell.
    this.updateCellsEmitter = new Emitter();
```
Seems like its documentation is very different from the name (ignoring "the the"). I've also seen things that say it "updates the view"? If it's an emitter that is only fired when it reaches its destination, then `reachedDestinationEmitter` would be more appropriate.

**Comment by jonathanolson** [comment]
Err sorry, there is a reachedDestinationEmitter just above it. How is reaching a cell different from reaching a destination (I guess if it's going back to a bucket?), and could we just use reachedDestinationEmitter?

**Comment by veillette** [comment]
We need to distinguish between the update of the cell and the piece reached its destination
The update of the fill of cell might have to be done before the animation, or after the animation depending if the piece emerge from the cell or ends on a cell. The removal of the piece from pieces array is always done at the end of the animation.

The current documentation for updateCellsEmitter is misleading but we need two separate Emitters.

**Comment by jonathanolson** [comment]
Is it possible to:

- If the cell would be 'updated' when the animation starts, just have whatever animates the piece also update the cell immediately.
- If the cell would be 'updated' when the animation ends, have whatever animates the piece add an onComplete listener that updates the cell when it ends?

That sounds preferable to a second emitter.

**Comment by veillette** [comment]
I still think we need two distinct emitters since we need to ask the piece to remove itself from the observable array, and to signal the containerSet has changed. I agree with you that there should only one emitter on the piece though. 

I think it would be preferable for the cellEmitter to be attached to the cell itself rather than the piece. 
In the long run. this would allow us to make very targeted updates on the view of ContainerSet.

**Comment by veillette** [comment]
On second thought, the emitter of cell is not necessary either since we have a property for isFilled.
We should leverage the isFilledProperty instead.

**Comment by veillette** [comment]
This issue is stale and no longer relevant. Closing
